doxia-core/src/main/java/org/apache/maven/doxia/macro/toc/TocMacro.java

```diff
@@ -112,9 +113,13 @@ public void execute(Sink sink, MacroRequest request) throws MacroExecutionExcept
             tocSink.close();
         }
 
-        IndexEntry index = tocSink.getRootEntry();
+        writeTocForIndexEntry(sink, getAttributesFromMap(request.getParameters()), tocSink.getRootEntry());
+    }
+
+    void writeTocForIndexEntry(Sink sink, SinkEventAttributes listAttributes, IndexEntry rootEntry) {
```

Comment:
This still needs some work to
a) only consider entries for type section
b) skip empty entries (not having an id)

Reply:
Done now.